src/libcmd/installables.cc

```diff
@@ -530,10 +530,12 @@ std::vector<InstallableValue::DerivationInfo> InstallableAttrPath::toDerivations
 
         std::set<std::string> outputsToInstall;
 
-        if (auto outputNames = std::get_if<OutputNames>(&extendedOutputsSpec))
+        auto * outputsSpec = std::get_if<ExtendedOutputsSpec::Explicit>(&extendedOutputsSpec);
```

Comment:
Is this variable used anywhere (other than in the `if`)?

Reply:
I rewrote this logic to hopefully be much easier to read.